src/player.h

```diff
@@ -356,11 +356,14 @@ class Player final : public Creature
 		return std::max<int32_t>(0, capacity - inventoryWeight);
 	}
 
-	int32_t getMaxHealth() const override { return std::max<int32_t>(1, healthMax + varStats[STAT_MAXHITPOINTS]); }
-	void setMaxHealth(int32_t newMaxHealth) override
+	int32_t getMaxHealth() const override
 	{
-		healthMax = newMaxHealth;
-		health = std::min<int32_t>(health, getMaxHealth());
+		return std::max<int32_t>(1, Creature::getMaxHealth() + varStats[STAT_MAXHITPOINTS]);
```

Comment:
This behavior seems incorrect when using `varStats`, because when using `maxHealth` in **Creature**., it will not retrieve the base value